runtime/src/lib.rs

```diff
@@ -683,6 +683,23 @@ sr_api::impl_runtime_apis! {
 		}
 	}
 
+	// TODO: What if we have multiple `set_heads` calls in the block?
+	impl parachain::ExtrinsicsQuerying<Block> for Runtime {
```

Comment:
This name is pretty bad.

Reply:
I wouldn't mind putting this into the `ParachainHost` API as `get_heads` or `extract_heads`